backend/app/app/api/api_v1/endpoints/ue_movement.py

```diff
@@ -514,53 +353,74 @@ def state_movement(
     """
     return {"running": retrieve_ue_state(supi, current_user.id)}
 
+
 @router.get("/state-ues", status_code=200)
 def state_ues(
     current_user: models.User = Depends(deps.get_current_active_user),
 ) -> Any:
     """
     Get the state
     """
-    return ues
+    return crud.ue.get_multi_by_owner(SessionLocal(), owner_id=current_user.id)
 
-#Functions
-def retrieve_ue_state(supi: str, user_id: int) -> bool: 
+
+# Functions
+def retrieve_ue_state(supi: str, user_id: int) -> bool:
     try:
         return threads[f"{supi}"][f"{user_id}"].is_alive()
     except KeyError as ke:
-        print('Key Not Found in Threads Dictionary:', ke)
+        print("Key Not Found in Threads Dictionary:", ke)
         return False
 
+
 def retrieve_ues() -> dict:
-    return ues
+    return crud.ue.get_multi_by_owner(SessionLocal(), owner_id=current_user.id)
+
 
 def retrieve_ue(supi: str) -> dict:
-    return ues.get(supi)
+    return crud.ue.get_supi(SessionLocal(), supi)
+
+
+def retrieve_ue_distances(supi: str, user_id: int) -> dict:
+    db = SessionLocal()
+    ue = crud.ue.get_supi(db, supi)
+    if ue is None:
+        return {}
+
+    cells = jsonable_encoder(crud.cell.get_multi_by_owner(db=db, owner_id=user_id))
+    _, distances = check_distance(ue.latitude, ue.longitude, cells)
```

Comment:
Check if it's easy to change `check_distance` to accept model directly, if so make it use it instead of using the `jsonable_encoder`. Don't forget to verify all call sites.